File: src/commands/adjustPoints.js
import { SlashCommandBuilder } from "discord.js";
import { ENV } from "../config/env.js";
import { supabase } from "../database/supabase.js";

export const adjustPoints = {
  data: new SlashCommandBuilder()
    .setName("adjust-points")
    .setDescription("Admin: Adjust points for a customer or chef")
    .addUserOption(option => 
      option.setName("user")
            .setDescription("User to adjust")
            .setRequired(true))
    .addIntegerOption(option =>
      option.setName("points")
            .setDescription("New points value")
            .setRequired(true))
    .addStringOption(option =>
      option.setName("reason")
            .setDescription("Optional reason for adjustment")
            .setRequired(false)),

  async execute(interaction) {
    const userId = interaction.user.id;

    // Admin check
    if (!ENV.ADMIN_ROLE_IDS.includes(userId)) {
      return interaction.reply({ content: "You are not allowed to use this command.", ephemeral: true });
    }

    const targetUser = interaction.options.getUser("user");
    const newPoints = interaction.options.getInteger("points");
    const reason = interaction.options.getString("reason") || "No reason provided";

    try {
      // Update points in Supabase (assuming a 'points' column in 'users' table)
      const { error } = await supabase
        .from("users")
        .update({ points: newPoints })
        .eq("id", targetUser.id);

      if (error) throw error;

      await interaction.reply({
        content: `✅ Points for ${targetUser.tag} updated to ${newPoints}. Reason: ${reason}`,
        ephemeral: true
      });
    } catch (err) {
      console.error("Adjust points error:", err);
      await interaction.reply({ content: "Failed to adjust points.", ephemeral: true });
    }
  }
};

File: src/commands/adjust_points.py
import nextcord
from nextcord import Interaction, Embed
from nextcord.ext import commands
from src.config.env import ADMIN_ROLE_IDS
from src.database.supabase import supabase

bot = commands.Bot(command_prefix="!")

@bot.slash_command(name="adjust-points", description="Admin only: Adjust points for a user")
async def adjust_points(
    interaction: Interaction,
    target_user: nextcord.Member,
    new_points: int,
    reason: str = "No reason provided"
):
    # Check if user is admin
    if not any(role.id in ADMIN_ROLE_IDS for role in interaction.user.roles):
        await interaction.response.send_message("❌ You are not allowed to use this command.", ephemeral=True)
        return

    # Update Supabase
    supabase.table("users").update({"points": new_points}).eq("id", str(target_user.id)).execute()

    embed = Embed(
        title="✅ Points Adjusted",
        description=f"Admin: {interaction.user.mention}\n"
                    f"User: {target_user.mention}\n"
                    f"New Points: {new_points}\n"
                    f"Reason: {reason}",
        color=0xe67e22
    )

    await interaction.response.send_message(embed=embed)
